wideq/client.py

```diff
@@ -349,8 +353,26 @@ def enum_name(self, key, value):
         """Look up the friendly enum name for an encoded value.
         """
         options = self.value(key).options
+        if value not in options:
+            logging.warning(
+                'Value `%s` for key `%s` not in options: %s. Values from API: '
+                '%s', value, key, options, self.data['Value'][key]['option'])
+            return _UNKNOWN
         return options[value]
 
+    def reference_name(self, key: str, value: Any) -> str:
+        """Look up the friendly name for an encoded reference value.
+
+        :param key: The referenced key.
+        :param value: The value whose name we want to look up.
+        :returns: The friendly name for the referenced value.
```

Comment:
…or `'-'` if the value is not found? (Or perhaps there's a deeper reason behind the inclusion of `-` that deserves documenting here?)